### ISSUE TITLE
C# -> VB: Symbols differing only in case should be renamed

### ISSUE THREAD

**Issue by RussellSmith2** [icsharpcode/CodeConverter#80]
This is the C#->VB counterpart of #103

### Input code
```C#
private static bool IsNotNullableReplacable(NameSyntax name, TypeSyntax reducedName)
{
    var isNotNullableReplacable = false;
}
```

### Erroneous output
```vbnet
Private Function IsNotNullableReplacable(ByVal name As NameSyntax, ByVal reducedName As TypeSyntax) As Boolean
	Dim isNotNullableReplacable = False
End Function
```

### Details
Local variables can't have the same name as the function the are in even if the case is different.

**Comment by GrahamTheCoder** [comment]
I think the best way to implement this might be as a separate pass on the input which finds and renames everything that clashes on case, hopefully making use of the existing [`Renamer`], (http://source.roslyn.codeplex.com/#Microsoft.CodeAnalysis.Workspaces/Rename/Renamer.cs,122757488c14f307,references
) then construct a compilation with the renamed output and proceed as normal.

**Comment by GrahamTheCoder** [comment]
Let's use this issue to cover everything about name collisions due to case insensitivity. I'll close 
 https://github.com/icsharpcode/CodeConverter/issues/221 and remember to consider type parameters when solving this.

**Comment by GrahamTheCoder** [comment]
A pre-conversion pass to detect case sensitive variants of the same name and call `Renamer.RenameSymbolAsync` would be a neat solution to this.

**Comment by ad48hp** [comment]
Probably not enough worthy addon to the discussion, but obviously this shall also apply when in a function, arguments that have same name as the fields of the specific class (which also can be inherited, makin' it more tricky..)..

Could ya look at this prob. soon enough too ?

**Comment by GrahamTheCoder** [comment]
Extra test cases always appreciated, thanks!

I've been putting this off for a long time because doing it *right* sounded hard, and I rarely see interest from users wanting to convert C# -> VB for a whole project (where this is most relevant), it's mainly just snippets.

However, I've made some architectural improvements and had some ideas that I think make this more feasible (as my reminder comment above hints), so I'll have a fresh look at how big the task seems now

**Comment by GrahamTheCoder** [comment]
Here's a rough outline of the solution:
https://github.com/icsharpcode/CodeConverter/compare/cs2vb/80/case-sensitivity?expand=1

Would still need a decent amount of time to get all the cases covered, and to try to get it vaguely performant. Anyone's welcome to pick this up and run with it (just post here and let me know). Otherwise I'll try to pick it up in the next month or so when I have time.

**Comment by GrahamTheCoder** [comment]
Sorry, I didn't get around to this in the window I had free, so it's back to being generically on the backlog.

**Comment by GrahamTheCoder** [comment]
The scaffolding for this is in place, just needs a finaly push [CaseConflictResolver.cs#L37](https://github.com/icsharpcode/CodeConverter/blob/master/ICSharpCode.CodeConverter/VB/CaseConflictResolver.cs#L37). Also see https://github.com/icsharpcode/CodeConverter/issues/487 for a special case of this.

**Comment by GrahamTheCoder** [comment]
Fixed in #505
